src/pdn/src/PdnGen.tcl

```diff
@@ -3149,65 +4500,61 @@ proc init {{PDN_cfg "PDN.cfg"}} {
   variable metal_layers
   variable def_units
   variable stripes_start_with
-  variable rails_start_with
   variable physical_viarules
   variable stdcell_area
   variable voltage_domains
+  variable global_connections
+  variable default_global_connections
+  variable power_nets
+  variable ground_nets
 
-#    set ::start_time [clock clicks -milliseconds]
-  init_tech
-
-  if {![file exists $PDN_cfg]} {
-    utl::error "PDN" 62 "File $PDN_cfg does not exist."
-  }
-
-  if {![file_exists_non_empty $PDN_cfg]} {
-    utl::error "PDN" 28 "File $PDN_cfg is empty."
-  }
+  # debug "start" 
+  init_tech 
 
   set design_name [$block getName]
 
-  set design_data {}
   set physical_viarules {}
   set stdcell_area ""
-
-  # debug "start"
-  source $PDN_cfg
-  write_pdn_strategy
-
+ 
   set die_area [$block getDieArea]
   utl::info "PDN" 8 "Design name is $design_name."
   set def_output "${design_name}_pdn.def"
 
   # debug "examine vars"
-  if {[info vars ::power_nets] == ""} {
-    set ::power_nets "VDD"
+  if {$power_nets == {}} {
+    if {[info vars ::power_nets] == ""} {
+      set power_nets "VDD"
+    } else {
+      set power_nets $::power_nets
+    }
   }
-
-  if {[info vars ::ground_nets] == ""} {
-    set ::ground_nets "VSS"
+  
+  if {$ground_nets == {}} {
+    if {[info vars ::ground_nets] == ""} {
+      set ground_nets "VSS"
+    } else {
+      set ground_nets $::ground_nets
+    }
   }
 
   if {[info vars ::core_domain] == ""} {
     set  ::core_domain "CORE"
```

Comment:
is there a reason this is getting set in the global namespace?

Reply:
I agree, shouldnt set a value in the global namespace. Fixed.